libnavigation-core/src/main/java/com/mapbox/navigation/core/MapboxNavigation.kt

```diff
@@ -449,6 +455,35 @@ constructor(
         tripSession.unregisterStateObserver(tripSessionStateObserver)
     }
 
+    /**
+     * Attach your own controller to determine when drivers arrived at stops via [ArrivalController]
+     * Use [navigateToNextStop] to manually move navigator to the next stop.
+     *
+     * @param arrivalController ArrivalObserver
+     */
+    fun attachArrivalController(arrivalController: ArrivalController) {
+        arrivalProgressObserver.attach(arrivalController)
+    }
+
+    /**
+     * Stop controlling arrival at stops, detaching will require the caller
+     * to call [navigateToNextStop] manually.
+     */
+    fun removeArrivalController() {
```

Comment:
```suggestion
    fun unattachArrivalController() {
```

or 

```kotlin
fun addArrivalController(arrivalController: ArrivalController) {
    arrivalProgressObserver.attach(arrivalController)
}
```

What do you think?

Reply:
hmm yeah the names were deliberate because because the behavior is different. but can see why the different names will just be confusing

Reply:
but then again, it really is different. can see from other comments why it may not be easy to understand at first glance